### PR TITLE
[stable/influxdb] Add backups to GCS

### PR DESCRIPTION
**Pull request by naseemkullah**
Signed-off-by: Naseem <naseemkullah@gmail.com>

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by naseemkullah** [comment]
bump

**Comment by naseemkullah** [comment]
cc @jackzampolin

**Comment by naseemkullah** [comment]
cc @Aisuko

File: stable/influxdb/templates/backup-cronjob.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{{- if .Values.backup.enabled }}
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: {{ template "influxdb.fullname" . }}-backup
```
Comment (Aisuko) [inline stable/influxdb/templates/backup-cronjob.yaml:5]: Should we add the annotations here?

File: stable/influxdb/templates/backup-cronjob.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  concurrencyPolicy: Forbid
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          restartPolicy: OnFailure
+          volumes:
+          - name: backups
+            emptyDir: {}
+          - name: google-cloud-key
```
Comment (Aisuko) [inline stable/influxdb/templates/backup-cronjob.yaml:28]: Is this just for GCP? Should we make a tip for deployer they can try other clouds provider? If the feature can only work for GCP should we add this to the master?

File: stable/influxdb/templates/backup-cronjob.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  name: {{ template "influxdb.fullname" . }}-backup
+  labels:
+    app: {{ template "influxdb.fullname" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+spec:
+  schedule: {{.Values.backup.schedule | quote }}
+  concurrencyPolicy: Forbid
+  jobTemplate:
```
Comment (Aisuko) [inline stable/influxdb/templates/backup-cronjob.yaml:16]: Should we add `metadata.labels` for `jobTemplate` and include `app: {{ template "influxdb.fullname" . }}` ,`release: "{{ .Release.Name }}"`

File: stable/influxdb/templates/backup-cronjob.yaml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    spec:
+      template:
```
Comment (Aisuko) [inline stable/influxdb/templates/backup-cronjob.yaml:18]: Same as `jobTemplate`

File: stable/influxdb/values.yaml [hunk 5]
```diff
@@ -262,3 +262,10 @@ initScripts:
   scripts:
     init.iql: |+
       CREATE DATABASE "telegraf" WITH DURATION 30d REPLICATION 1 NAME "rp_30d"
+
+backup:
+  enabled: false
+  schedule: "0 0 * * *"
+  destination: gs://bucket/influxdb
+  google:
```
Comment (Aisuko) [inline stable/influxdb/values.yaml]: Should we make a change to `google`? e.g gcp or gcloud?  I believe can make sense better.

**Review by Aisuko — CHANGES_REQUESTED** [review]
Thanks for your PR, this was absolutely good work. There may have some advice for the PR.

File: stable/influxdb/templates/backup-cronjob.yaml (hunk 1 above)
Comment (naseemkullah) [reply stable/influxdb/templates/backup-cronjob.yaml:5]: Sure, added.

File: stable/influxdb/templates/backup-cronjob.yaml (hunk 3 above)
Comment (naseemkullah) [reply stable/influxdb/templates/backup-cronjob.yaml:16]: Sure, added.

File: stable/influxdb/templates/backup-cronjob.yaml (hunk 4 above)
Comment (naseemkullah) [reply stable/influxdb/templates/backup-cronjob.yaml:18]: Not here I think

File: stable/influxdb/templates/backup-cronjob.yaml (hunk 2 above)
Comment (naseemkullah) [reply stable/influxdb/templates/backup-cronjob.yaml:28]: For now it is just for GCS right. I encourage others to add for other storage backends.

File: stable/influxdb/values.yaml (hunk 5 above)
Comment (naseemkullah) [reply stable/influxdb/values.yaml]: yep I went with gcs (Google Cloud Storage)

File: stable/influxdb/Chart.yaml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: influxdb
-version: 1.2.1
+version: 1.2.2
```
Comment (Aisuko) [inline stable/influxdb/Chart.yaml]: Should we bump version to 1.3.0? The PR was a new feature and in a backward-compatible manner?

**Review by Aisuko — CHANGES_REQUESTED** [review]

File: stable/influxdb/templates/backup-cronjob.yaml (hunk 1 above)
Comment (Aisuko) [reply stable/influxdb/templates/backup-cronjob.yaml:5]: Thanks

File: stable/influxdb/Chart.yaml (hunk 6 above)
Comment (naseemkullah) [reply stable/influxdb/Chart.yaml]: true!

**Pull request merged**